backend/src/models/Application.ts

```diff
@@ -0,0 +1,46 @@
+import { InferSchemaType, Schema, model } from "mongoose";
+import { applicationStatusSchema } from "./ApplicationStatus";
+
+const applicationSchema = new Schema({
+  _id: {
+    type: String,
+    unique: true,
+    required: true,
+  },
+  userId: {
+    type: String,
+    required: true,
+  },
+  companyId: {
+    type: Schema.Types.ObjectId,
+    required: true,
+  },
+  companyName: {
+    type: String,
+    required: true,
+  },
+  position: {
+    type: String,
+    required: true,
+  },
+  link: {
+    type: String,
+    required: false,
+  },
+  progress: {
+    type: [applicationStatusSchema],
+    default: [],
+  },
+  dateCreated: {
+    type: Date,
+    default: Date.now,
+  },
+  dateModified: {
+    type: Date,
+    default: Date.now,
+  },
```

Comment:
Apologies for the confusion, I meant for the `dateCreated` and `dateModified` fields to be the default `timestamps` fields (`createdAt` and `modifiedAt`) so feel free to change this to `timestamps`. I will make sure to update the backend doc! And also FYI I think you should be able to modify the PR by just simply pushing to the same branch in your remote repo without a request to change.